x-pack/plugins/security_solution/public/detections/containers/detection_engine/rules/use_pre_packaged_rules_status.ts

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { useCallback } from 'react';
+import { useQuery, useQueryClient } from 'react-query';
+import { useAppToasts } from '../../../../common/hooks/use_app_toasts';
+import { getPrePackagedRulesStatus } from './api';
+import * as i18n from './translations';
+
+const ONE_MINUTE = 60000;
+
+export interface PrePackagedRulesStatusResponse {
+  rulesCustomInstalled: number;
+  rulesInstalled: number;
+  rulesNotInstalled: number;
+  rulesNotUpdated: number;
+  timelinesInstalled: number;
+  timelinesNotInstalled: number;
+  timelinesNotUpdated: number;
+}
+
+export const PRE_PACKAGED_RULES_STATUS_QUERY_KEY = 'prePackagedRulesStatus';
+
+export const usePrePackagedRulesStatus = () => {
+  const { addError } = useAppToasts();
+
+  return useQuery<PrePackagedRulesStatusResponse>(
+    [PRE_PACKAGED_RULES_STATUS_QUERY_KEY],
+    async ({ signal }) => {
+      const response = await getPrePackagedRulesStatus({ signal });
+
+      return {
+        rulesCustomInstalled: response.rules_custom_installed,
+        rulesInstalled: response.rules_installed,
+        rulesNotInstalled: response.rules_not_installed,
+        rulesNotUpdated: response.rules_not_updated,
+        timelinesInstalled: response.timelines_installed,
+        timelinesNotInstalled: response.timelines_not_installed,
+        timelinesNotUpdated: response.timelines_not_updated,
+      };
+    },
+    {
+      staleTime: ONE_MINUTE * 15,
```

Comment:
As with the installedIntegrations cache expiration, there will be some UX implications here in instances where we can't invalidate the cache, like when a new/updated integration makes additional rule assets available. 

For instance, if you install the prepackaged rules, then go to fleet and install the `LotL Attack Detection` integration (which has a few security rule assets bundled in), then go back to the Rule Mgmt page, you won't see the `Install X Elastic prebuilt rule(s)` button until you perform a mutation that invalidates the cache or this stale timeout is hit. 

Again, probably not a huge deal, especially with the added benefit of not always hitting this API so much, but definitely something the user is going to notice, so maybe we dial this one back to 5min or something? What do you think?

Reply:
Agree, reduced that time to 5 mins as well.